### ISSUE TITLE
Export to Import failing on several legacy applications

### ISSUE THREAD

**Issue by temujin9** [flynn/flynn#3351]
`error reading slug response: malformed HTTP version "Detected"` or `error reading slug response: malformed HTTP status code "environment"` are the results.

debug-info:
- https://gist.github.com/99c069f7e7bbd01bc9d4c53bc2c2fcc7 
- https://gist.github.com/2ca1f86aad523821d14d27cd9cacb708
- https://gist.github.com/4286df34b9d2c0dd16927501056883cc
- https://gist.github.com/1978e33091413f445b487354d13dc495
- https://gist.github.com/90a43769c6315a2d1ff04f6ac9024257

**Comment by lmars** [comment]
I suspect this is due to the deprecated `Cmd` / `Entrypoint` change (looks like a slug is being built rather than downloading a slug).

Investigating.

**Comment by lmars** [comment]
@temujin9 what version of the Flynn CLI are you using to do the import / export (`flynn version`), and what is the cluster version you are exporting from?

**Comment by temujin9** [comment]
Flynn CLI version: v20160830.0

Flynn cluster versions: v20160827.0 and v20160830.0 (this has shown up on two clusters, now)

**Comment by temujin9** [comment]
Sorry for introducing ambiguity: the debug info is from the v20160830.0 cluster.

**Comment by temujin9** [comment]
Issue is ongoing with a v20160906.0 source cluster, v20160907.0 sink cluster and v20160907.0 flynn binary.

**Comment by lmars** [comment]
@temujin9 sorry for the slow response on this one.

So from the error you are getting (i.e. `error reading slug response: ...`), I assume this is an error you are getting running `flynn export`? If so, could you perhaps gist the output of `flynn release show --json`?

Also, are you using external blobstore storage (e.g. S3)?

**Comment by temujin9** [comment]
I am not using external blobstore.

I thought that error was from the import, but sure enough it is the export.

```
{
  "id": "a3df1545-cfa0-47d8-b1bd-62362a366b14",
  "artifacts": [
    "356f806b-8fb7-4997-9966-375e1f1a6781",
    "926deb6f-dd33-4bdb-8458-2943b5d97d40"
  ],
  "env": {
    "FLYNN_APP_NAME": "firebird-teamcity-radiator",
    "FLYNN_APP_SCOPE": "firebird",
    "FLYNN_CLUSTER_NAME": "flynn-ci-us-east-1",
    "FLYNN_NEXUS_REGION": "us-east-1",
    "FLYNN_NEXUS_UNIVERSE": "ci",
    "SLUG_URL": "http://blobstore.discoverd/91888de0-2a4c-4edd-a3f7-bf68ff2c23fa.tgz"
  },
  "meta": {
    "git": "true"
  },
  "processes": {
    "web": {
      "args": [
        "/runner/init",
        "start",
        "web"
      ],
      "ports": [
        {
          "port": 8080,
          "proto": "tcp",
          "service": {
            "name": "firebird-teamcity-radiator-web",
            "create": true,                                                                                                                                                                                                                                             
            "check": {                                                                                                                                                                                                                                                  
              "type": "tcp"                                                                                                                                                                                                                                             
            }                                                                                                                                                                                                                                                           
          }                                                                                                                                                                                                                                                             
        }                                                                                                                                                                                                                                                               
      ],                                                                                                                                                                                                                                                                
      "resources": {                                                                                                                                                                                                                                                    
        "cpu": {                                                                                                                                                                                                                                                        
          "request": 1000,                                                                                                                                                                                                                                              
          "limit": 1000                                                                                                                                                                                                                                                 
        },                                                                                                                                                                                                                                                              
        "max_fd": {                                                                                                                                                                                                                                                     
          "request": 10000,                                                                                                                                                                                                                                             
          "limit": 10000                                                                                                                                                                                                                                                
        },                                                                                                                                                                                                                                                              
        "memory": {                                                                                                                                                                                                                                                     
          "request": 1073741824,                                                                                                                                                                                                                                        
          "limit": 1073741824                                                                                                                                                                                                                                           
        }                                                                                                                                                                                                                                                               
      },                                                                                                                                                                                                                                                                
      "cmd": [                                                                                                                                                                                                                                                          
        "start",                                                                                                                                                                                                                                                        
        "web"                                                                                                                                                                                                                                                           
      ],                                                                                                                                                                                                                                                                
      "entrypoint": [                                                                                                                                                                                                                                                   
        "/runner/init"                                                                                                                                                                                                                                                  
      ]                                                                                                                                                                                                                                                                 
    }                                                                                                                                                                                                                                                                   
  },                                                                                                                                                                                                                                                                    
  "created_at": "2016-06-22T20:41:47.058458Z",                                                                                                                                                                                                                          
  "artifact": "356f806b-8fb7-4997-9966-375e1f1a6781"                                                                                                                                                                                                                    
}                                                                                                                                                                                                                                                                       ```

**Comment by temujin9** [comment]
(Additionally, the export hangs for a long time after that, and refuses to die with Ctrl-C, only dying when I background and `kill -9` it.)

**Comment by lmars** [comment]
Ok thanks, can you also run this:

```
$ flynn -a controller pg psql -- -c "SELECT uri FROM artifacts WHERE artifact_id = '926deb6f-dd33-4bdb-8458-2943b5d97d40'"
```

**Comment by temujin9** [comment]
Sorry for the delay.

```
                                   uri                                    
--------------------------------------------------------------------------
 http://blobstore.discoverd/52803ee1-b210-4203-b01e-9158f51cb434/slug.tgz
(1 row)
```

**Comment by temujin9** [comment]
Import & export is the currently the best way to work around the root cause of #3413 in the short term, making this a higher priority.

**Comment by lmars** [comment]
@temujin9 ok, so does that slug exist? Try running:

```
$ flynn run curl -s -o /dev/null -w '%{http_code}\n' http://blobstore.discoverd/52803ee1-b210-4203-b01e-9158f51cb434/slug.tgz
```

It should print 200.

**Comment by temujin9** [comment]
Yep, it did:
```
Detected 512 MB available memory, 512 MB limit per process (WEB_MEMORY)
Recommending WEB_CONCURRENCY=1
200
```

**Comment by lmars** [comment]
@temujin9 aha! That output before the 200 is what is causing the issue, I assume this is a Nodejs app?

**Comment by lmars** [comment]
So to get this app to export, you should unset the `LOG_CONCURRENCY` environment variable and then try running the export again.

I'll come up with a proper fix.

**Comment by temujin9** [comment]
`LOG_CONCURRENCY` is not set (per `flynn env`).

**Comment by lmars** [comment]
@temujin9 how about `flynn env -t web`? I am working on a proper patch now anyway.

**Comment by temujin9** [comment]
```
FLYNN_APP_NAME=firebird-teamcity-radiator
FLYNN_APP_SCOPE=firebird
FLYNN_CLUSTER_NAME=flynn-ci-us-east-1
FLYNN_NEXUS_REGION=us-east-1
FLYNN_NEXUS_UNIVERSE=ci
SLUG_URL=http://blobstore.discoverd/91888de0-2a4c-4edd-a3f7-bf68ff2c23fa.tgz
```

(Those FLYNN_ are our internal info.)

**Comment by temujin9** [comment]
What if I explicitly set `LOG_CONCURRENCY=false`?

**Comment by lmars** [comment]
@temujin9 I suspect the app in question was built using an old version of the Nodejs buildpack which just prints those lines regardless of `LOG_CONCURRENCY` (i.e. before [this commit](https://github.com/heroku/heroku-buildpack-nodejs/commit/459ebb67baff09c62fe10d24d9b530915b1da722)).

I have proposed a fix in #3419, are you able to wait until that is released, restore to a new cluster and then do the export? If not I can write a program which will export the legacy app.

**Comment by lmars** [comment]
@temujin9 fixed merged in #3419, should make it into a release tonight.

**Comment by temujin9** [comment]
Wonderful, thanks!

**Comment by temujin9** [comment]
That fixed it; thanks again.

**Comment by temujin9** [comment]
This is back, as of v20161115.0.

**Comment by temujin9** [comment]
Correction: it is back as of v20161102.0. I forgot the problem was in the exporting cluster.

**Comment by lmars** [comment]
@temujin9 what version of the CLI did you use to create the export? Are you getting exactly the same errors?

**Comment by temujin9** [comment]
CLI version is automatically updated, so it was probably v20161115.0 at the time.

**Comment by lmars** [comment]
@temujin9 fixed (again) in #3703.
